Pseudo/css/pseudoElements.css

```diff
@@ -0,0 +1,101 @@
+/*FirstLetter*/
+
+.firstLetter p::first-letter {
+	color: #ff0000;
+	font-size: 23px;
+}
+
+/*firstLine*/
+
+.firstLine p::first-line {
+	color: #00ff00;
+}
+
+/*selection*/
+
+.content p::selection {
+	background-color: #ffff00;
+}
+
+/*beforeafterStyling*/
+
+.beforeafterStyling {
+	text-align: center;
+	overflow-x: hidden;
+}
+.beforeafterStyling h3{
+	text-align: left;
+}
+.beforeafterStyling h4 {
+	display: inline-block;
+	position: relative;
+}
+.beforeafterStyling h4::before,.beforeafterStyling h4::after {
+	position: absolute;
+	content: '';
+	width: 900px;
+	height: 2px;
+	background-color: #ff0000;
+	top: 10px;
+}
+.beforeafterStyling h4::before {
+	right: 100%;
+	margin-right: 15px;
+}
+.beforeafterStyling h4::after {
+	left: 100%;
+	margin-left: 15px;
+}
+
+/*hoverEffect*/
+
+.hoverEffect a {
+	border: 1px solid #000;
+	padding: 10px 15px;
+	display: inline-block;
+	position: relative;
+}
+.hoverEffect a::before,.hoverEffect a::after {
+	width: 15%;
+	height: 40%;
+	content: '';
+ 	position: absolute;
+ 	transition: all 0.4s ease-in-out;
+}
+.hoverEffect a::before {
+	border-left: 1px solid #000000;
+	border-top: 1px solid #000000;
+	left:0;
+	top: 0;
+}
+.hoverEffect a::after {
+ 	border-right: 1px solid #000000;
+ 	border-bottom: 1px solid #000000;
+ 	right: 0;
+ 	bottom: 0;
+}
+.hoverEffect a:hover:before,.hoverEffect a:hover::after {
```

Comment:
Why have you used :before and not ::before? Please comment the valid reason behind it, please don't push the code with ::before prior to commenting. Also have a habit of adding a space after ,

Reply:
Already discussed at Brota.